app/scripts/lib/experiment.js

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'lib/experiments/open-gmail',
+  'lib/experiments/mailcheck',
+  'lib/experiments/coppa-view',
+  'lib/url'
+], function (_, OpenGmailExperiment, MailcheckExperiment, CoppaExperiment, Url) {
+  'use strict';
+
+  var CHOOSE_ABLE_EXPERIMENT = 'chooseAbExperiment';
+  var FORCE_EXPERIMENT_PARAM = 'forceExperiment';
+  // all available experiments that must be independent for A/B testing purposes.
+  var EXPERIMENTS = {
```

Comment:
Do you ever foresee other types of experiments being mutually exclusive, e.g., on the signin or settings pages? I am trying to think through how this can be made more generic and some of the work done in either experiments-mixin or a subclass of experiments-mixin.

What I'm thinking is that the ExperimentMixin (or subclass) could pass in the name of the able experiment to use in place of [CHOOSE_ABLE_EXPERIMENT](https://github.com/mozilla/fxa-content-server/pull/2943/files#diff-4d13c3e741a80fc8e5c551f515b822e2R65), as well as a factory function.

Able would make the choice, return a string which indicates which experiment to run, the string is passed to the factory function. The factory function creates the experiment, returns it, which is then tracked here.

That would reduce the number of items [passed in to the ExperimentInterface constructor](https://github.com/mozilla/fxa-content-server/pull/2943/files#diff-4d13c3e741a80fc8e5c551f515b822e2R29) to only those items needed to make the choice.

Different ExperimentMixins could be created where it makes sense, perhaps a SignUpExperimentMixin,  SignInExperimentMixin, and a SettingsExperimentMixin.